test/utils/floating_point_utils.hpp

```diff
@@ -42,21 +53,40 @@ struct FP
     /// See: https://webassembly.github.io/spec/core/syntax/values.html#canonical-nan.
     static constexpr auto canon = UintType{1} << (num_mantissa_bits - 1);
 
-    T value{};
+    UintType value{};
+
+    FP() = default;
 
-    explicit FP(T v) noexcept : value{v} {}
+    FP(FloatType v) noexcept : value{bit_cast<UintType>(v)} {}
 
-    explicit FP(UintType u) noexcept : value{bit_cast<T>(u)} {}
+    explicit FP(UintType u) noexcept : value{u} {}
+
+    operator TypedValue() const noexcept
+    {
+        if constexpr (std::is_same_v<FloatType, float>)
+            return {ValType::f32, Value{value}};
+        else
+            return {ValType::f64, Value{value}};
+    }
 
     /// Return unsigned integer with the binary representation of the value.
-    UintType as_uint() const noexcept { return bit_cast<UintType>(value); }
+    UintType as_uint() const noexcept { return value; }
 
-    /// Returns NaN payload if the value is a NaN, otherwise 0 (NaN payload is never 0).
-    UintType nan_payload() const noexcept
+    /// Return the floating-point value.
+    FloatType as_float() const noexcept { return bit_cast<FloatType>(value); }
+
+    /// Returns true if the value is a NaN.8
+    bool is_nan() const noexcept
     {
-        return std::isnan(value) ? (as_uint() & mantissa_mask) : 0;
+        const auto u = as_uint();
```

Comment:
Why `std::is_nan` was not enough?

Reply:
Yep, an explanation in a comment would be nice.

Reply:
This is now explained in the comments.